api/pkg/common/common.go

```diff
@@ -176,6 +178,35 @@ func InitCtxWithAuthInfo(request *restful.Request) context.Context {
 	return ctx
 }
 
+// TODO: Can be optimized with above function InitCtxWithAuthInfo()
+func InitCtxWithAuthInfoNonAdmin(request *restful.Request) context.Context {
+	actx := request.Attribute(c.KContext).(*c.Context)
+	var ctx context.Context
+	if actx.IsAdmin {
+		ctx = metadata.NewContext(context.Background(), map[string]string{
+			CTX_KEY_USER_ID:   actx.UserId,
+			CTX_KEY_TENANT_ID: actx.TenantId,
+			CTX_KEY_IS_ADMIN:  strconv.FormatBool(true),
+		})
+	} else {
+		ctx = metadata.NewContext(context.Background(), map[string]string{
+			CTX_KEY_TENANT_ID: request.PathParameter("tenantId"),
+			CTX_KEY_IS_ADMIN:  strconv.FormatBool(false),
+		})
+	}
+
+	return ctx
+}
+
+func GetAdminContext() context.Context {
```

Comment:
Why this function? InitCtxWithAuthInfo() should suffice, right?

Reply:
If calls are coming from user, context are created for user. So, in tiering case ...normal user can not list backends, so adminCtx required